textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslatorTest.java

```diff
@@ -10,6 +12,16 @@
 
 public class RegexToGramQueryTranslatorTest {
 	
+	private void printTranslatorResult(String regex) {
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		
+		System.out.println("regex: "+regex);
```

Comment:
A few thoughts on this:

- We definitely need automated test cases.  It's OK to create a boolean query **manually**, even if we create it based on the output of the translator (done offline once).  I don't view it as "cheating."  The value is there when we make changes to other modules, and this test case can serve as a "security guard."

- We shouldn't use "RegexMatcher" to test this translator.  

- Testing logical equivalence of two boolean expressions is a hard problem.  The current implementation is based on tree homomorphism, which is not complete.  That is, two boolean expressions can still be logically equivalent even if they are not homomorphic to each other.  Can we translate both to DNF, then do the testing?

- In terms of implementation, we don't need to implement all the optimizations in one PR.  Let's have one version that's correct and do the merge.  Then implement other improvements as separate PRs.  For instance, removing redundancy can be done separately.